core/trino-main/src/main/java/io/trino/sql/planner/planprinter/JsonRenderer.java

```diff
@@ -54,6 +83,88 @@ private JsonRenderedNode renderJson(PlanRepresentation plan, NodeRepresentation
                         .collect(toImmutableList()));
     }
 
+    public Map<Integer, List<DistributedPlanRepresentation>> render(PlanPrinter planPrinter, boolean verbose, Integer level)
+    {
+        PlanRepresentation plan = planPrinter.getPlanRepresentation();
+        NodeRepresentation node = plan.getRoot();
+        Map<Integer, List<DistributedPlanRepresentation>> distributedPlanRepresentationMap = new HashMap<>();
+        distributedPlanRepresentationMap.computeIfAbsent(level, i -> new ArrayList<>()).add(getPlanRepresentation(plan, node, verbose, level));
+        return distributedPlanRepresentationMap;
+    }
+
+    public DistributedFragmentRepresentation formatFragment(
```

Comment:
This is also misnamed. The method is about turning plan fragment into a DistributedFragmentRepresentation